### PR TITLE
Fixed #29517 -- Added support for SQLite column check constraints on positive integer fields.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/29517

### CODE REVIEW

File: django/db/backends/sqlite3/base.py [hunk 1]
```diff
@@ -76,6 +76,10 @@ class DatabaseWrapper(BaseDatabaseWrapper):
         'TimeField': 'time',
         'UUIDField': 'char(32)',
     }
+    data_type_check_constraints = {
+        'PositiveIntegerField': '"%(column)s" >= 0',
+        'PositiveSmallIntegerField': '"%(column)s" >= 0',
+    }
```
Comment (charettes) [inline django/db/backends/sqlite3/base.py]: I guess this is required because `unsigned` isn't enforced by SQLite at all?

Comment (timgraham) [reply django/db/backends/sqlite3/base.py]: Correct

**Comment by timgraham** [comment]
Is the release note what you had in mind?

File: docs/releases/2.2.txt [hunk 2]
```diff
@@ -218,7 +218,10 @@ Database backend API
 Miscellaneous
 -------------
 
-* ...
+* On SQLite, ``PositiveIntegerField`` and ``PositiveSmallIntegerField`` now
+  include a check constraint to prevent negative values in the database. If you
+  have existing invalid data and run a migration that recreates a table, you'll
+  see ``CHECK constraint failed``.
```
Comment (charettes) [inline docs/releases/2.2.txt]: @timgraham it looks good to me!

**Pull request merged**